powerauth-java-server/src/test/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/ActivationServiceBehaviorTest.java

```diff
@@ -120,8 +136,19 @@ void testPrepareActivationWithInvalidPayload() throws Exception {
         // Prepare activation with missing devicePublicKey
         final String activationCode = initActivationResponse.getActivationCode();
         final String applicationKey = detailResponse.getVersions().get(0).getApplicationKey();
-        final GenericServiceException exception = assertThrows(GenericServiceException.class, () ->
-                tested.prepareActivation(activationCode, applicationKey, false, encryptedRequest, version, keyConvertor));
+        final GenericServiceException exception = assertThrows(GenericServiceException.class, () -> {
+            final PrepareActivationRequest request = new PrepareActivationRequest();
+            request.setActivationCode(activationCode);
+            request.setGenerateRecoveryCodes(false);
+            request.setProtocolVersion(version);
+            request.setApplicationKey(applicationKey);
+            request.setMac(encryptedRequest.getMac());
+            request.setNonce(encryptedRequest.getNonce());
+            request.setEncryptedData(encryptedRequest.getEncryptedData());
+            request.setEphemeralPublicKey(encryptedRequest.getEphemeralPublicKey());
+            request.setTimestamp(encryptedRequest.getTimestamp());
+            tested.prepareActivation(request);
+        });
```

Comment:
The request could be created out of the lambda.
```suggestion
        final PrepareActivationRequest request = new PrepareActivationRequest();
        request.setActivationCode(activationCode);
        request.setGenerateRecoveryCodes(false);
        request.setProtocolVersion(version);
        request.setApplicationKey(applicationKey);
        request.setMac(encryptedRequest.getMac());
        request.setNonce(encryptedRequest.getNonce());
        request.setEncryptedData(encryptedRequest.getEncryptedData());
        request.setEphemeralPublicKey(encryptedRequest.getEphemeralPublicKey());
        request.setTimestamp(encryptedRequest.getTimestamp());

        final GenericServiceException exception = assertThrows(GenericServiceException.class, () ->
                tested.prepareActivation(request));
```

Reply:
OK